### PR TITLE
Add `wcadmin_settings_change` tracks event to shipping fields

### LINKED ISSUE #34519: `wcadmin_settings_change` not fired when adding/updating shipping zones/classes
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When either adding or updating a shipping class or zone within **WooCommerce > Settings > Shipping** the `wcadmin_settings_change` track is not being triggered. 

### Expected behavior

`wcadmin_settings_change` track event should be triggered when adding/updating shipping zones and classes.

### Actual behavior

Only the `wcadmin_settings_view` is triggered for the shipping zones and classes tab, but no `settings_change` track is being triggered.
`wcadmin_settings_change` is being triggered when updating the shipping options under **WooCommerce > Settings > Shipping**.

### Steps to reproduce

1. Install WooCommerce and [WooCommerce beta tester](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-beta-tester)
2. Make sure logging is enabled under **WooCommerce > Settings > Advanced > WooCommerce.com**
3. Go to **WooCommerce > Settings > Shipping** and add a new shipping zone and a new shipping class
4. Go to **WooCommerce > Status > Logs** and select the log starting with `tracks-2022-`.... for today's date (this is added by the WooCommerce beta tester)
5. Notice how only the `wcadmin_settings_view` track is triggered for the shipping zone and class pages but no `wcadmin_settings_change`
6. Try updating a shipping option under **WooCommerce > Settings > Shipping > Shipping options** and check the logs again, notice how `wcadmin_settings_change` is triggered now.

### WordPress Environment

WooCommerce version: 6.9.0
WordPress: 6.0.1

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by nathanss**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

I created a new action in the tracking class called `woocommerce_update_non_option_setting ` to make it work for shipping zones, shipping methods and shipping classes, since the class only worked with options that end up calling the `update_option` hook, which is not the case for these settings, so they need to be handled differently.

Closes #34519 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure logging is enabled under WooCommerce > Settings > Advanced > WooCommerce.com
2. Go to WooCommerce > Settings > Shipping and add a new shipping zone
4. Fill the shipping zone fields: zone name, zone regions, post codes, add some shipping methods, and disable some shipping methods. Press 'Save changes'.
5. Go to WooCommerce > Status > Logs and select the log starting with tracks-2022-.... for today's date (this is added by the WooCommerce beta tester)
6. Check all the logged 'wcadmin_settings_change' events. There should be different 'wcadmin_settings_change' events with attributes 'zone_method', 'zone_name,zone_locations,zone_postcodes', 'zone_method_settings', and 'zone_methods,zone_methods_enabled'.
7. Re-order two shipping methods and press 'Save changes'.
8. There should be a new event with attribute 'zone_methods,zone_methods_order'.
9. Go to WooCommerce > Settings > Shipping and go to Shipping classes tab.
10. Add a new shipping class and fill name, slug, and description.
8. There should be a new event with attribute 'shipping_class_name,shipping_class_slug,shipping_class_description'.

### CODE REVIEW

File: plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php [hunk 1]
```diff
@@ -59,10 +59,16 @@ class WC_Settings_Tracking {
 	public function init() {
 		add_action( 'woocommerce_settings_page_init', array( $this, 'track_settings_page_view' ) );
 		add_action( 'woocommerce_update_option', array( $this, 'add_option_to_list' ) );
+		add_action( 'woocommerce_update_shipping_setting', array( $this, 'add_option_to_list_and_track_change' ) );
```
Comment (mattsherman) [inline plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php]: I imagine there are other settings like this, that don't use options under the hood?

I'd make this action name more generic.

What about something like `woocommerce_update_non_option_setting`? I know it it a bit wordy, but it seems clear to me. 😄

File: plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php [hunk 2]
```diff
@@ -59,10 +59,16 @@ class WC_Settings_Tracking {
...
 		add_action( 'woocommerce_update_options', array( $this, 'send_settings_change_event' ) );
 		add_action( 'admin_enqueue_scripts', array( $this, 'possibly_add_settings_tracking_scripts' ) );
 	}
 
+	public function add_option_to_list_and_track_change( $option ) {
```
Comment (mattsherman) [inline plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php]: What about `add_option_to_list_and_track_setting_change` for consistency with other function names?

File: plugins/woocommerce/includes/class-wc-ajax.php [hunk 3]
```diff
@@ -3030,8 +3030,14 @@ public static function shipping_zone_add_method() {
 
 		$zone_id     = wc_clean( wp_unslash( $_POST['zone_id'] ) );
 		$zone        = new WC_Shipping_Zone( $zone_id );
+		do_action( 'woocommerce_update_shipping_setting', array( 'id' => 'zone_method' ) );
 		$instance_id = $zone->add_shipping_method( wc_clean( wp_unslash( $_POST['method_id'] ) ) );
 
+
+		global $current_tab;
+		$current_tab = 'shipping';
```
Comment (mattsherman) [inline plugins/woocommerce/includes/class-wc-ajax.php:3066]: Oh globals. I guess that since the current implementation depends on this, it is okay. 😄

**Comment by mattsherman** [comment]
> I created a new action in the tracking class called `woocommerce_update_shipping_setting` to make it work for shipping zones, shipping methods and shipping classes, since the class only worked with options that end up calling the `update_option` hook, which is not the case for these settings, so they need to be handled differently. Let me know what you think about this implementation and if you think of any alternatives.

I think the general approach seems reasonable. I'd probably make it generic though.

> I'm not sure if we should track values as well, since most of them are quite complex. It's currently not tracking the values.

I wouldn't track any values that are user input. If they are boolean or an option from a hard-coded list, etc. I would.

File: plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php (hunk 1 above)
Comment (nathanss) [reply plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php]: I like it!

File: plugins/woocommerce/includes/class-wc-ajax.php (hunk 3 above)
Comment (nathanss) [reply plugins/woocommerce/includes/class-wc-ajax.php:3066]: Yeah... I don't like it as well

**Review by mattsherman — APPROVED** [review]
Tested and works well! Nice work.

One thing I did notice, that wasn't clearly specified anywhere... we are not logging Tracks events when the following occur:
* Shipping zone deleted
* Shipping class deleted

Those should be done in a follow-up PR.
